[Wayne Tan] iP

tanwayne890 opened the pull request.

Commit:
Add Gradle support

tanwayne890 committed:
add Level-1

tanwayne890 committed:
add level - 2

tanwayne890 committed:
add level3

tanwayne890 committed:
add level-4

tanwayne890 committed:
add A-TextUiTesting

tanwayne890 committed:
add level-5

tanwayne890 committed:
add level-6

tanwayne890 committed:
corrected level-5

tanwayne890 committed:
Add more OOP and Storage

tanwayne890 committed:
Debug OOP

tanwayne890 committed:
Merge branch 'branch-Level-7'

tanwayne890 committed:
Add and debug storage

tanwayne890 committed:
Add date and time format

tanwayne890 committed:
Merge branch 'master' into branch-Level-7

tanwayne890 committed:
Merge branch 'branch-Level-7'

tanwayne890 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Command.java
#	src/main/java/Duke.java
#	src/main/java/Storage.java

tanwayne890 committed:
Add JUnit Tests for Task and Deadline class

tanwayne890 committed:
Package the App as a JAR file & Add data directory to gitignore

tanwayne890 committed:
Add JavaDoc comments

tanwayne890 committed:
Tweak the code to comply with a coding standard

tanwayne890 committed:
Add find keyword

tanwayne890 committed:
Merge branch 'branch-A-JavaDoc'

tanwayne890 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Parser.java
#	src/main/java/TaskList.java

tanwayne890 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Command.java
#	src/main/java/CommandType.java

tanwayne890 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

tanwayne890 committed:
Add Gradle for automated project build and set up gradle to run unit test

tanwayne890 committed:
Merge branch 'branch-A-Gradle'

tanwayne890 committed:
Add JavaFX by following the guide

tanwayne890 committed:
Remove printing Ui messages to console.

Duke GUI need string to print out messages.

Let's,
* Make UI to return string when command execute
* Add nessesary methods in UI to return string for specific command

siddarth2824 commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,263 @@
+     * @return boolean
+     */
+    public boolean isExit() {
+        if (commandType.equals(CommandType.EXITDUKE)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
```
I do not think this is a coding violation but I think a way you could try to implement to shorten this code would be to `return commandType.equals(CommandType.EXITDUKE)`. This would just be a suggestion to shorten the code and make it simpler for others to read as well.

tanwayne890 replied:
wow cool, i could not think about this 👍  Thanks!

siddarth2824 commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,263 @@
+            InvalidDeadlineDescripDukeException, InvalidEventDescripDukeException,
+            InvalidFirstDukeException, ParseException, IOException {
+
+        if (commandType.equals(CommandType.PRINTALLTASKS)) {
+            printAllTask(tasks);
+        } else if (commandType.equals(CommandType.EXITDUKE)) {
+            ui.closeDuke();
+        } else if (commandType.equals(CommandType.MARKASDONE)){
+            int counter = Integer.parseInt(commandArr[1]);
+            tasks.getTask(counter - 1).markAsDone();
+            storage.saveToFile(tasks.tasks);
+        } else if (commandType.equals(CommandType.DELETETASK)){
+            int counter = Integer.parseInt(commandArr[1]);
+            deleteTask(counter - 1, tasks);
+            storage.saveToFile(tasks.tasks);
+        } else if (commandType.equals(CommandType.ADDTODO)){
+            tasks.addTask(createTodo());
+            storage.saveToFile(tasks.tasks);
+        } else if (commandType.equals(CommandType.ADDDEADLINE)) {
+            tasks.addTask(createDeadline());
+            storage.saveToFile(tasks.tasks);
+        } else if (commandType.equals(CommandType.ADDEVENT)) {
+            tasks.addTask(createEvent());
+            storage.saveToFile(tasks.tasks);
+        } else if (commandType.equals(CommandType.FINDTASK)) {
+            String keyword = joinString(removeFirst(commandArr));
+            ArrayList<Task> tempTasks = searchKeyWord(tasks, keyword);
+            printSearchedTask(tempTasks);
+        } else {}
```
There is no coding violation here but would it better if you change the multiple else-if statements to a switch statement and handle each different case within the switch statement? This might help to improve code readability and make the code even cleaner.

tanwayne890 replied:
Thanks for the reminder

siddarth2824 commented on src/main/java/CommandType.java:
```diff
@@ -0,0 +1,4 @@
+public enum CommandType {
+    EXITDUKE, PRINTALLTASKS, ADDTODO, ADDDEADLINE,
+    ADDEVENT, MARKASDONE, DELETETASK, FINDTASK;
```
According to the Java coding standard guide, under the Naming section, it says that constants should be all uppercase using underscore to separate words. Should there be an underscore between variables with multiple words in it? For example, `EXITDUKE` could instead be `EXIT_DUKE`?

jeffreytjs replied:
I agree, actually it is possible to keep the names concise like EXITDUKE could be understood the same way as EXIT, the ADD and TASK within the names could be removed and still be interpreted in a similar manner. There is nothing wrong with keeping your names explicit but it would have to follow the naming convention as mentioned by Siddarth above.

tanwayne890 replied:
Thanks for the tips!!

siddarth2824 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,24 @@
+// a task that need to specify a date to be done before the date
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
```
According to the CS2103/T textbook, under the "Guideline: Name well" section in the Code Quality chapter, it says that nouns should be used for classes/variables. Maybe you could consider using a noun to describe your variable. You could maybe use `deadlineDate` or `dateTime` instead of `by`.

tanwayne890 replied:
thanks for reminding. i actually forgot about it

siddarth2824 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,24 @@
+// a task that need to specify exact date and time
+public class Event extends Task {
+
+    protected LocalDateTime at;
```
According to the CS2103/T textbook, under the "Guideline: Name well" section in the Code Quality chapter, it says that nouns should be used for classes/variables. Maybe you could consider using a noun to describe your variable. You could maybe use `eventTime` or `dateTime` instead of  `at`.

siddarth2824 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,63 @@
+        String[] commandArr = fullCommand.split(" "); // split input into string array
+        Command command;
+
+        if (commandArr[0].equals("bye")) {
+            command = new Command(CommandType.EXITDUKE, null);
+        } else if (commandArr[0].equals("list")) {
+            command = new Command(CommandType.PRINTALLTASKS, null);
+        } else if (commandArr[0].equals("done")) {
+            command = new Command(CommandType.MARKASDONE, commandArr);
+        } else if (commandArr[0].equals("delete")) {
+            command = new Command(CommandType.DELETETASK, commandArr);
+        } else if (commandArr[0].equals("todo")) {
+            command = new Command(CommandType.ADDTODO, commandArr);
+        } else if (commandArr[0].equals("deadline")) {
+            command = new Command(CommandType.ADDDEADLINE, commandArr);
+        } else if (commandArr[0].equals("event")) {
+            command = new Command(CommandType.ADDEVENT, commandArr);
+        } else if (commandArr[0].equals("find")) {
+            command = new Command(CommandType.FINDTASK, commandArr);
+        } else {
+            throw new InvalidFirstDukeException();
+        }
+        return command;
```
There is no coding violation here but would it better if you change the multiple else-if statements to a switch statement and handle each different case within the switch statement? This might help to improve code readability and make the code even cleaner.

siddarth2824 commented on src/main/java/StorageParser.java:
```diff
@@ -0,0 +1,50 @@
+public class StorageParser {
+
+    public static boolean isDone(String symbol) {
+        if (symbol.equals("✓")) {
+            return true;
+        } else {
+            return false;
+        }
```
I do not think this is a coding violation but I think a way you could try to implement to shorten this code would be to `return symbol.equals(icon)`. This would just be a suggestion to shorten the code and make it simpler for others to read as well.

siddarth2824 reviewed (commented):
Good job on following the java coding guide. Your code is very easy to read for me and its very clean. Your class, variable and method naming is very clear and purposeful. Great job overall.

tanwayne890 committed:
Debug GUI to show tick and cross symbol properly

Lets,
* Add FileWriter's charsets UTF-8
* Add Storage's scanner charsets UTF-8

tanwayne890 committed:
Merge branch 'branch-Level-10'

tanwayne890 committed:
Add more date and time input format for Deadline and Event

jeffreytjs commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,242 @@
+import java.io.IOException;
```
Missing package. Every class should be part of some package according to convention.

jeffreytjs commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,242 @@
+    /**
+     * Command constructor specifying CommandType and command array.
+     *
+     * @param commandType
```
I like that some of your @param and @return contain the type, could try to include some simple description like  @param commandType Command from user input.

Not sure if it's meaningful to include params if it's self explanatory.

tanwayne890 replied:
ok noted, will include description if it is not self explanatory

jeffreytjs commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    public static void main(String[] args) {
+            String[] temp = {"2019-12-20", "18:00"};
```
Is this meant to be hard coded for formatting purpose?

tanwayne890 replied:
yes for testing

jeffreytjs reviewed (commented):
Overall, neat coding standards you have there. Some codes could be simplified and do look into the naming conventions again! Keep up the good work with the exploration of javafx! 👍

tanwayne890 committed:
Create jar file using Gradle shadow

tanwayne890 committed:
Add assertions to Command, Parser and Storage

tanwayne890 committed:
Merge pull request #2 from tanwayne890/branch-A-Assertions

Add assertions to Command, Parser and Storage

tanwayne890 committed:
Add javafx-sdk to gitignore

tanwayne890 committed:
Delete javafx sdk

tanwayne890 committed:
Edit code after PR review

- Add switch branch
- Rename enum type
- Make variable more understandable

tanwayne890 committed:
Create gradle.yml

tanwayne890 committed:
Modify gradle.yml

tanwayne890 committed:
Update gradle.yml

Ignore IO test

tanwayne890 committed:
Add Gradle checkstyle to IDEA

tanwayne890 committed:
Merge branch 'master' of https://github.com/tanwayne890/ip

tanwayne890 committed:
Modify code to follow checkstyle rule

tanwayne890 committed:
Modify code to follow checkstyle rule

tanwayne890 committed:
Debug TaskTest

tanwayne890 committed:
Improve Code Quality

Let's,
- Implement SLAP
- reduce long method

tanwayne890 committed:
Merge pull request #3 from tanwayne890/branch-A-CodeQuality

Improve Code Quality

tanwayne890 committed:
Add Date and Description Comparator

tanwayne890 committed:
Add sort by date and sort by description commands

tanwayne890 committed:
Merge pull request #4 from tanwayne890/branch-Sort

Add sorting by date command and sorting by description command

tanwayne890 committed:
Improve Duke GUI

tanwayne890 committed:
Add create directory method for first time user.

tanwayne890 committed:
Add user guide

tanwayne890 committed:
Edit user guide

*Add duke logo

tanwayne890 committed:
Debug index from the command exceeds tasks list size

*Introduce the exception
*Throw the exception when get the task outside of tasks list size

tanwayne890 committed:
Add Javadoc for IndexExceedException

tanwayne890 committed:
Handle sensitive symbols for input

*Add new exception
*Add new method to check the input string

tanwayne890 committed:
Add exceptions for date format & invalid deadline and event format

* Add date format exception handling to Parser
* Add deadline and event exception handling to Command

tanwayne890 committed:
Edit readme.md

tanwayne890 committed:
Remove unused import
